### ISSUE TITLE
A few questions

### ISSUE THREAD

**Issue by kahoona77** [smezger/somfy-signalduino#1]
Hi,
I just found this repository and I am very excited about it because I am searching for something to control my somfy blinds with the signladuino but without FHEM. My goal would be to have something that can be included into Home Assistant.

I tried to get it working on my machine, but had no luck yet. So I have a couple of questions, if you don't mind:

- From the example-paths like `/dev/serial/by-id/usb-Unknown_radino_CC1101-if00` I am assuming you are running on Linux or Mac (with .net-core?). Currently I trying to get it working under Windows. Did you run it on Windows? If yes, did you use any "special" drivers?
- Is there any way to get logs or debug-outputs from the signalduino, because I am seeing nothing there?

Currently the signalduino is working with FHEM and I can control the somfy blinds. I also copied the RollingCodes and the EncryptionKey from there.

It would be realy nice, if you could give some hints.
Thanks in advance,
Benjamin

**Comment by smezger** [comment]
@kahoona77 
Let me try to answer your questions/support you. First of all I have to make clear the project state, I just started work on this a couple of weeks ago and did not find the time to test it on real blinds!
* The code in general should work on win as well but haven't tested it, There should be no need of special drivers, jsut use the correct COM port, that can be found in Windows "Device Manager" and put e.g. "COM2" in.
* regarding logs from signalduino I have no idea if one can get logs. You can check fi the communication itself is working by issuing e.g. "V" (Version) or "P" (Ping) command and check for the response.

If you have copied RC and EncKey from FHEM I can give some hints at least. FHEM stores the last used values, while as my implementation requires the values used for the next transmission. --> you have to increment both values first, when copying from FHEM.
All values in FHEM are hex values like RC, EncKey, DeviceAddress, thus make sure when you use them to specify them as hex.
I hope this helps. Let me know! I will also check if I can find some spare time in the next days to debug it on real blinds.

**Comment by smezger** [comment]
@kahoona77 
Hi Benjamin,
I've just pushed a new version that is working on my side, check: 56bc307.
Hope that helps! 😸

**Comment by kahoona77** [comment]
Hi Seabstian,

thanks for your help! I got it woking now after the update 👍 

I had to install the drivers from here: https://www.nanocul.de/selfsite.php?aktion=Firmware. Without them Windows did not recognize the USB-device and did not assign it to a COM-port.
Also I had some Issues with the rolling-codes. After a few tries I got lucky and found the next rolling-code. After that it was working like a charm :)

Now I will check how I will integrate it in Home Assistant. Maybe I will port it to python or golang for that. I am more familar there than in .net.

Thanks again for your help!
